openssh: some options aren't supported by 'configure' anymore

PeterPawn opened the pull request:
- unknown options for 'configure':
$ ./configure --enable-static --enable-shared --disable-debug
configure: WARNING: unrecognized options: --enable-static, --enable-shared, --disable-debug

- static linking didn't work as expected, needs 'libdl.a' library in the right order

er13 commented:
What is the libdl-order-change needed for? I have no problems compiling openssh statically in different configurations (with/without openssl), with all openssh binaries, with just some of them.

PeterPawn commented:
I've no idea anymore ... it's too long ago, that I've tried and found it necessary.

If it can be linked without problems in your build host, it should be enough - if I find later some time to start an own build, I'll try it without this exlicit order once more, too.

PS: Thx for the commits.

PeterPawn commented:
I was unable to reproduce the case, where I ran into the problem with library order.

I would guess, that I had done a first build without explicit selection of the OpenSSH package (only with "sftp-server" as supplemental for "dropbear") and with OpenSSL as crypto provider - a reference from "libssh" to "libdl" is only then plausible ... and possibly it was a "culture clash" later, after I changed only the "-static" option - but the rebuild dependency on this static option (and the option at all) was introduced a long time ago and I would expect, that a fresh build took place after I had changed the option.

But as far as I remember (and that's not really good), from somewhere (I would guess, from "openbsd-compat") another function with the same name as one from "libdl" was provided. I don't know precisely its name anymore ... but the list of possibilities isn't really huge.

And to circumvent linking with the wrong function, I've changed this order to ensure, that "libdl" is the immediate successor of "libssh" in the search order. Maybe the effect was only visible, if the binary was used later and it tried to load an OpenSSL DSO - but I'm really sorry, I can't remember reliably and as long as I can't reproduce it, this change is obviously superfluous.

It's a little bit strange ... if I were not ran into this problem, I had never taken a deeper look onto this package and the removed options for 'configure' would never caught my attention.

Anyway ... I'll close this PR myself now - clearly a case of "works for me". :-)

er13 commented:
I could imagine you had a problem with one of the older versions of openssh. In the meantime it has been fixed upstream. That's the reason we both are not able to reproduce it anymore.
